client/js/utils.js

```diff
@@ -63,12 +63,12 @@ function scrollIntoViewNicely(el) {
 
 function collapse() {
 	$(".chan.active .toggle-button.toggle-preview.opened").click();
-	return true;
+	return "";
```

Comment:
What is this change for?
Really this function shouldn't return anything, it's weirder that it returns `""`, but it's already weird that it returns anything imo.

Reply:
It returned `true` before to block sending the command to the server. This PR changes it to return a string that should be sent to the server (in this case empty string = don't send anything).

We need returns to selectively block sending it to the server.

Reply:
Should this change or good to go?

Reply:
Sorry, I didn't see @xPaw's answer earlier. It's honestly very weird. `collapse()` and `expand()` should really not return anything.

It's bizarre right now, because we have 3 commands including 2 that do not go to the server, and the third one returns a string, so we have them all return a string, then check if output is a string and if the string is empty...

I suggest removing the return values from both `expand` and `collapse` functions. Then your code above could be:

```js
if (typeof utils.inputCommands[cmd] === "function") {
	const output = utils.inputCommands[cmd](args);

	if (output === undefined) {
		return false;
	}

	text = output;
}
```

Would this work? I feel like it wouldn't because of the other branch in `join` (when L75 of this file is `undefined`).

Reply:
This will kind of work - the problem with your code is `/join #chan` doesn't get sent to the server at all (as far as I can tell), so /join is now broken.

Reply:
@astorije poke for when you can get to it

Reply:
You are correct that my naive solution is incorrect.
I still think we should refactor a little bit to avoid having a `return ""` to represent "this command is local and should not go to the server". This is just too much error-prone and we _will_ break this soon enough (silently or not).

Maybe @xPaw's suggestion to combine with #2004 could help making this a bit nicer?